#!/usr/bin/env python
PACKAGE = "amrl_motion_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#gen.add("learning_rate",                double_t, 0, "Learning rate for background subtraction", 0.5, 0, 1);
gen.add("history",                      int_t   , 0 , "History frame of background subtractor", 300, 50, 700);
gen.add("erosion_iterations",                      int_t   , 0 , "The iteration number of erosion filter", 2, 1, 10);
gen.add("dilation_iterations",                      int_t   , 0 , "The iteration number of dilation filter", 2, 1, 10);
gen.add("learning_rate",                    double_t   , 0 , "Negative parameter value makes the algorithm to use some automatically chosen learning rate.\
 0 means that the background model is not updated at all,\
 1 means that the background model is completely reinitialized from the last frame.", -1,-1,1);
gen.add("shadow",                      bool_t   , 0 , "History frame of background subtractor",True);
gen.add("min_area",int_t,0,"min area of conutors",500,0,20000)
gen.add("morphology_kernel_size",int_t,0,"morphology kernel size nxn", 3,2,10);
gen.add("dilation_kernel_size",int_t,0,"morphology kernel size nxn", 3,2,10);


exit(gen.generate(PACKAGE, "motion_detection_node", "MotionDetection"))
